rfcs/0029-backports-team.md

```diff
@@ -0,0 +1,251 @@
+merged PRs from github and provide them for the "figuring out" task previously
+described. The PRs would be presented from oldest to newest.
+
+Once filed, the collaboration system will allow members to assign themselves
+to do the backports. It is expected that team members will go from oldest to
+newest, but not enforced. Working on implementing a backport would first have
+the team member "take" the task, allowing parallel work to not conflict.
+
+A task will be worked on by making use of a command line tooling, allowing
+most tediousness to be automated away.
+
+### Making the backport
+
+As previously said, the team member will use a command line tooling.
+
+The team member will be presented with the commands in the interface, allowing
+less thoughts to be given to the task.
+
+When using the tooling, pretty much all tasks will be as automated as possible.
+
+ * Fetching
+ * (Checking out current-stable)
+ * Creating the branch
+ * Cherry-picking
+     * If it fails to auto-apply, this is where work starts; applying.
+     * After applying, resuming automation will be possible.
+ * Build of affected attributes (*may be limited somewhat*)
+
+One part that is **not** automated is testing. It is expected that the team
+member will at least try to launch affected software `result/bin/*`. Only a
+general glance would be required.
+
+For patches affecting libraries, affected attributes will be listed, allowing
+some general testing to be done.
+
+Once some general testing has been done and conclusive, the team member will
+use the tooling to automate pushing and creating the PR.
+
+ * Push to team member's repository.
+ * Create the PR targeting current-stable.
+     * The text will be auto-filled with data according to the build.
+     * Some machine-usable data may be provided for the tooling.
+ * Automatically mark the task as done and add with the newly opened PR number to it.
+
+The standardized PRs created by the tooling are expected to be easier to be
+approve by members of NixOS. The reduced number of manual steps is expected
+to make it easy for non-maintainers, even only enthusiasts to take part in
+the effort.
+
+
+Drawbacks
+=========
+[drawbacks]: #drawbacks
+
+Formalizing a process always reeks of red tape. This may turn off some
+volunteers away from doing backports.
+
+It may well happen that once the team formed, nothing changes and the backports
+are not maintained any more than they are.
+
+Writing and maintaining tooling may slow down or halt progress toward actually
+maintaining backports.
+
+
+Alternatives
+============
+[alternatives]: #alternatives
+
+Continuing as-it-is, with ad-hoc updates, sometimes missed. This has proved not
+to be the most successful way to manage backports.
+
+It could also be possible to only implement parts of the RFC. Either the team
+or the tooling. Both are of equal value and generally independent. They would,
+though, work best if working together.
```

Comment:
It would be possible to add "should be backported" as a check-box on PRs, automating the discovery part.